backend/uclapi/dashboard/models.py

```diff
@@ -35,6 +36,30 @@ class App(models.Model):
     created = models.DateTimeField(auto_now=False, auto_now_add=True)
     last_updated = models.DateTimeField(auto_now=True)
 
+    client_id = models.CharField(
+        max_length=33,
+        unique=True,
+        default=generate_app_client_id
+    )
+
+    client_secret = models.CharField(
+        max_length=64,
+        unique=True,
+        default=generate_app_client_secret
+    )
+
+    callback_url = models.CharField(max_length=500, default="")
```

Comment:
Sure, does this explain your choice? https://stackoverflow.com/a/10052288/5297057